src/CompositeContainer.php

```diff
@@ -44,8 +44,14 @@ public function has($id)
      * Attaches a container to the composite container.
      * @param ContainerInterface $container
      */
-    public function attach(ContainerInterface $container): void
+    public function attach(ContainerInterface $container, bool $parentLookup = false): void
```

Comment:
A common name for it is "delegate lookup".

```suggestion
    public function attach(ContainerInterface $container, bool $delegateLookup = false): void
```

Also, a phpdoc is needed. It should explain that when set to true, dependencies will be looked for in composite container instead of in $container directly. When set to false, dependencies would be looked for in $container only.

Reply:
Done